xskillscore/core/comparative.py

```diff
@@ -0,0 +1,151 @@
+            Larger values (positive). Defaults to 'negative'. Ignored if
+            ``metric== categorical``.
+
+        Returns
+        -------
+        xarray.DataArray or xarray.Dataset reduced by dim containing the sign test and
+            confidence as new dimension results:
+
+                * ``sign_test``: positive (negative) number shows how many times over
+                    ``dim`` ``forecast1`` is better (worse) than ``forecast2``.
+                * ``confidence``: Positive boundary for the random walk at significance
+                    ``alpha``.
+
+
+        Examples
+        --------
+        >>> f1 = xr.DataArray(np.random.normal(size=(30)),
+        ...      coords=[('time', np.arange(30))])
+        >>> f2 = xr.DataArray(np.random.normal(size=(30)),
+        ...      coords=[('time', np.arange(30))])
+        >>> o = xr.DataArray(np.random.normal(size=(30)),
+        ...      coords=[('time', np.arange(30))])
+        >>> st = sign_test(f1, f2, o, dim='time', metric='mae', orientation='negative')
+        >>> st.sel(results='sign_test').plot()
+        >>> st.sel(results='confidence').plot(c='gray')
+        >>> (-1*st.sel(results='confidence')).plot(c='gray')
+
+        References
+        ----------
+            * DelSole, T., & Tippett, M. K. (2016). Forecast Comparison Based on Random
+              Walks. Monthly Weather Review, 144(2), 615–626. doi: 10/f782pf
+    """
+
+    def _categorical_metric(observations, forecasts, dim):
+        """Returns True where forecasts exactly equals observations"""
+        return observations == forecasts
+
+    if metric is not None:
+        # make sure metric is a callable
+        if isinstance(metric, str):
+            if metric == 'categorical':
+                metric = _categorical_metric
+                if orientation != 'positive':
+                    warnings.warn(
+                        'Changing to "orientation=positive" for consistency with "metric=categorical"',
+                        UserWarning,
+                    )
+                orientation = 'positive'
+            else:
+                import xskillscore as xs
+
+                if hasattr(xs, metric):
+                    metric = getattr(xs, metric)
+                else:
+                    raise ValueError(
+                        f'xskillscore.metric could not be derived from {metric}.'
+                    )
+        elif not callable(metric):
+            raise ValueError(
+                f'metric needs to be a function/callable or None, found {type(metric)}'
+            )
+        if observation is not None:
+            # Compare the forecasts and observation using metric
+            metric_f1o = metric(observation, forecast1, dim=[])
+            metric_f2o = metric(observation, forecast2, dim=[])
+        else:
+            raise ValueError(
+                'observations must be provided when metric is provided', UserWarning
```

Comment:
What is `UserWarning` here for?

Reply:
What so you mean? Don’t understand your comment

Reply:
UserWarning not needed. deleted